source/common/common/hash.h

```diff
@@ -14,6 +15,22 @@ class HashUtil {
    * See https://github.com/Cyan4973/xxHash for details.
    */
   static uint64_t xxHash64(absl::string_view input) { return XXH64(input.data(), input.size(), 0); }
+
+  /**
+   * TODO(gsagula): extend xxHash to handle case-insensitive.
+   *
+   * Return 64-bit hash representation of string ingnoring case.
+   * See djb2 (http://www.cse.yorku.ca/~oz/hash.html) for more details.
+   * @param input supplies the string view.
+   * @return 64-bit hash representation of the supplied string view.
+   */
+  static uint64_t djb2CaseInsensitiveHash(absl::string_view input) {
+    uint64_t hash = 5381;
+    for (unsigned char c : input) {
+      hash += ((hash << 5) + hash) + absl::ascii_tolower(c);
```

Comment:
Seems like an unintended deviation from the original DJB2 hash having:
```
      hash += ((hash << 5) + hash) + absl::ascii_tolower(c);
```
instead of:
```
      hash = ((hash << 5) + hash) + absl::ascii_tolower(c);
```
While at it, a typo in `ingnoring`.

Reply:
Thanks for your answer. Please check #21787.